process_qsos_dr7.m

```diff
@@ -0,0 +1,476 @@
+        log_priors_no_c4(quasar_ind) + log_likelihoods_no_c4(quasar_ind);
+    
+    % fprintf(' ... log p(D | z_QSO, no CIV)     : %0.2f\n', ...
+        % log_likelihoods_no_c4(quasar_ind));
+    
+    min_z_c4s(quasar_ind) = min_z_c4(this_wavelengths);
+    max_z_c4s(quasar_ind) = max_z_c4(z_qso, max_z_cut);
+    
+    sample_z_c4 = ...
+        min_z_c4s(quasar_ind) +  ...
+        (max_z_c4s(quasar_ind) - min_z_c4s(quasar_ind)) * offset_z_samples;
+
+
+    % Temperature samples
+    
+    % ensure enough pixels are on either side for convolving with
+    % instrument profile
+
+   % building a finer wavelength and mask arrays 
+   % by adding the mean of ith and ith +1 element
+    lenW = length(this_wavelengths);
+        
+    fine_wavelengths = finer(this_wavelengths, nAVG);
+    fine_pixel_mask = finer(this_pixel_mask, nAVG);
+    fine_pixel_mask(fine_pixel_mask>0)=1;
+    if(Rsl==1)
+        fine_sigma_pixel = finer(this_sigma_pixel, nAVG); 
+    end
+    
+    this_rest_fine_wavelengths = emitted_wavelengths(fine_wavelengths, z_qso);
+    unmasked_ind_fine = (this_rest_fine_wavelengths >= min_lambda) & ...
+    (this_rest_fine_wavelengths <= max_lambda);
+    this_unmasked_fine_wavelengths = fine_wavelengths(unmasked_ind_fine);
+
+    padded_fine_wavelengths = ...
+        [logspace(log10(min(this_unmasked_fine_wavelengths)) - width * pixel_spacing/2, ...
+        log10(min(this_unmasked_fine_wavelengths)) - pixel_spacing/2,...
+        width)';...
+        this_unmasked_fine_wavelengths';...
+        logspace(log10(max(this_unmasked_fine_wavelengths)) + pixel_spacing/2,...
+        log10(max(this_unmasked_fine_wavelengths)) + width * pixel_spacing/2,...
+        width)'...
+        ];
+    
+    % [mask_ind] to retain only unmasked pixels from computed absorption profile
+    % this has to be done by using the unmasked_ind which has not yet
+    % been applied this_pixel_mask.
+    ind = (~this_pixel_mask(unmasked_ind));
+    
+    % compute probabilities under DLA model for each of the sampled
+    % (normalized offset, log(N HI)) pairs
+    lenW_unmasked = length(this_unmasked_wavelengths);
+    parfor i = 1:num_C4_samples
+        
+        % Limitting red-shift in the samples
+        this_z_c4 = (this_wavelengths / 1549) - 1;
+        % % absorption corresponding to this sample with one absorption line as a noise model 
+
+                
+        % absorptionL1 = absorptionL1(ind);
+        % c4_muL1     = this_mu     .* absorptionL1;
+        % c4_ML1      = this_M      .* absorptionL1;
+        % %     dla_omega2 = this_omega2 .* absorption.^2;
+        
+        % sample_log_likelihoods_c4L1(quasar_ind, i) = ...
+        %     log_mvnpdf_low_rank(this_flux, c4_muL1, c4_ML1, ...
+        %     this_noise_variance);
+
+        % absorption corresponding to this sample with two absorption lines as a doublet 
+        % compute fine absorption 
+        num_lines=2;
+        if(Rsl==1)
```

Comment:
What's this condition for?

Reply:
Rsl is  a handle. If it's 1 it means the code uses the version of voigt.c that also includes pixel resolution (sigma_pixels) from SDSS spectra during the convolution process in voigt.c. If it's 0 it uses the simple version of voigt.c

Reply:
I see! Thanks!